server/expression.js

```diff
@@ -173,33 +173,34 @@ expression.readMatrix_ = function(buf) {
 
 /**
  * Reads the TFA matrix data from the given buffer.
- * @param {!Buffer} buf Buffer of the TFA matrix data file.
+ * @param {string} fileName Name of the TFA matrix data file.
  * @return {!expression.RawMatrix} The TFA matrix data as a JS object.
  * @private
  */
-expression.readTFAmat_ = function(buf) {
-  var result = {};
-  var offset = 0;
-  var n = buf.readInt32LE(0);
-  var m = buf.readInt32LE(4);
-  var lrows = buf.readInt32LE(8);
-  var lcols = buf.readInt32LE(12);
-  offset += 16;
-  var rowstr = buf.toString('utf8', offset, offset + lrows); offset += lrows;
-  var colstr = buf.toString('utf8', offset, offset + lcols); offset += lcols;
-  result.numrows = n;
-  result.numcols = m;
-  result.rownames = rowstr.split(' ');
-  result.colnames = colstr.split(' ');
-  result.values = [];
-  for (var i = 0; i < n; i++) {
-    for (var j = 0; j < m; j++) {
-      var val = buf.readDoubleLE(offset);
-      offset += 8;
-      result.values.push(val);
-    }
+expression.readTFAmat_ = function(fileName) {
+  var lines = fs.readFileSync(fileName).toString().split('\n');
+  var parts = lines[0].split(/[\t\s]+/);
```

Comment:
Please assume TFA to be with matrix text format. The original data was in R and I wrote a script to convert it to the .rawdata format you're parsing here. However the .rawdata format is now obsolete and we will switch to the tsv format you see for expression matrix. 

See my email for details.